Sources/rapidsnark/rapidsnark.swift

```diff
@@ -3,201 +3,198 @@
 
     var errorMessageBuffer: [CChar] = Array(repeating: 0, count: errorBufferSize)
 
-    let result = groth16_verify(
+    // Call the rapidsnark C++ library function to perform the Groth16 Verification
+    let statusCode = groth16_verify(
         proofBuf,
         inputsBuf,
         verificationKeyBuf,
         &errorMessageBuffer,
         UInt(errorBufferSize)
     );
 
-    if (result == VERIFIER_VALID_PROOF) {
+    if (statusCode == VERIFIER_VALID_PROOF) {
         return true;
     }
-
-    let error = String(cString: errorMessageBuffer)
-
-    if (result == VERIFIER_INVALID_PROOF) {
-        throw RapidsnarkVerifierError.invalidProof(message: error)
-    } else if (result == VERIFIER_ERROR) {
-        throw RapidsnarkVerifierError.error(message: error)
-    }
-
-    throw RapidsnarkUnknownStatusError()
+    
+    throw groth16proverStatusCodeErrors(statusCode, message: String(cString: errorMessageBuffer))
 }
 
 public func groth16PublicSizeForZkeyBuf(
     zkey: Data,
-    errorBufferSize: Int = DEFAULT_ERROR_BUFFER_SIZE
+    errorBufferSize: Int = defaultErrorBufferSize
 ) throws -> Int {
     let zkeyBuffer = NSData(data: zkey).bytes
 
     var errorMessageBuffer: [CChar] = Array(repeating: 0, count: errorBufferSize)
 
-    var publicSize = 0
+    var size = 0
 
     let statusCode = groth16_public_size_for_zkey_buf(
         zkeyBuffer,
         UInt(zkey.count),
-        &publicSize,
+        &size,
         &errorMessageBuffer,
         UInt(errorBufferSize)
     );
 
     if (statusCode == PROVER_OK) {
-        return publicSize
-    } else {
-        let error = String(cString: errorMessageBuffer)
-        throw RapidsnarkProverError.error(message: error)
+        return size
     }
+    
+    throw RapidsnarkProverError.error(message: String(cString: errorMessageBuffer))
 }
 
+
+// Determine the necessary buffer size for storing public signals
+// based on a provided .zkey file
 public func groth16PublicSizeForZkeyFile(
     zkeyPath: String,
-    errorBufferSize: Int = DEFAULT_ERROR_BUFFER_SIZE
+    errorBufferSize: Int = defaultErrorBufferSize
 ) throws -> Int {
     var errorMessageBuffer: [CChar] = Array(repeating: 0, count: errorBufferSize)
 
-    var publicSize = 0
+    var size = 0
 
     let statusCode = groth16_public_size_for_zkey_file(
         zkeyPath,
-        &publicSize,
+        &size,
         &errorMessageBuffer,
         UInt(errorBufferSize)
     );
 
     if (statusCode == PROVER_OK) {
-        return publicSize
-    } else {
-        let error = String(cString: errorMessageBuffer)
-        throw RapidsnarkProverError.error(message: error)
+        return size
+    }
+    
+    throw RapidsnarkProverError.error(message: String(cString: errorMessageBuffer))
+}
+
+// Helper function to convert the status code and message into a specific error type
+private func groth16proverStatusCodeErrors(_ statusCode: Int32, message: String) -> RapidsnarkError {
+    switch statusCode {
+    case PROVER_ERROR:
```

Comment:
If these are all the errors we're expecting, I would wrap them in an enum with an associated value.  Something like
```
enum Groth16Result: Int {
case proverError = PROVER_ERROR
...
}
```
The advantage to this is that the compiler will let you know if there's a result value you aren't handling. You can create this from our C results with 
```
let result = Groth16Result(rawValue: resultFromC)
```
This returns an Optional, nil if you get something unexpected back.

Since nobody outside this file cares about the Groth16Result, you can label it `fileprivate` and define that here.

Reply:
Per discussion, since these values are from C headers, this would be tricky to implement.